indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -348,6 +348,44 @@ public boolean registerUpgradedPendingSegmentOnSupervisor(
     return false;
   }
 
+  public boolean changeTaskCountSupervisor(String id, Integer taskCount)
```

Comment:
This method needs a javadoc,  you can move the javadoc of the internal method here instead.

```suggestion
  public boolean updateTaskCountForSupervisor(String supervisorId, Integer taskCount)
```